plugins/modules/net_tools/nmcli.py

```diff
@@ -1065,6 +1064,9 @@ def _compare_conn_params(self, conn_info, options):
                     current_value = [re.sub(r'^{\s*ip\s*=\s*([^, ]+),\s*nh\s*=\s*([^} ]+),\s*mt\s*=\s*([^} ]+)\s*}', r'\1 \2 \3',
                                      route) for route in current_value]
                     current_value = [re.sub(r'^{\s*ip\s*=\s*([^, ]+),\s*nh\s*=\s*([^} ]+)\s*}', r'\1 \2', route) for route in current_value]
+                if key == self.mac_setting:
+                    # MAC addresses are case insensitive, nmcli always reports them in uppercase
+                    value = str.upper(value)
```

Comment:
Out of curiosity, why not:
```suggestion
                    value = value.upper()
```
?

(Also, why not also converting `current_value` to upper case?)

Reply:
`value.upper()` should be fine. `current_value` however comes from `conn_info`. That's already uppercase since that's the output of `nmcli connection show`.

Reply:
Oh you mean convert current_value into uppercase if `nmcli` at some point decides to change behaviour?

Reply:
Exactly. Not that I think that will happen, but better safe than sorry ;)